src/routes/synapse/room_events.rs

```diff
@@ -419,7 +442,15 @@ async fn update_friendship_status(
         }
     };
 
-    let room_event = serde_json::to_string(&room_info.room_event).unwrap();
+    let room_event_string = match serde_json::to_string(&room_info.room_event) {
+        Ok(room_event_string) => room_event_string,
+        Err(err) => {
+            log::error!("Error serializing room event: {:?}", err);
+            let _ = transaction.rollback().await;
+            return Err(SynapseError::CommonError(CommonError::Unknown));
+        }
+    };
+    let room_event = Box::leak(room_event_string.into_boxed_str());
```

Comment:
Can you handle it in another way? I mean, Are you sure you want to do this? Because you are explicitly saying "I'll leak this memory" 
Checkout: https://doc.rust-lang.org/std/boxed/struct.Box.html#method.leak

Reply:
@lauti7 I'm glad you mention this cause I noticed that the `leak` thing was not a good idea haha, but I don't know what I can do to handle it another way 🥲. Maybe we can meet and see it together?